rpcwebsocket.go

```diff
@@ -975,6 +981,16 @@ func (c *wsClient) handleMessage(msg []byte) {
 		return
 	}
 
+	// Check if the user is read-only and disconnect client if unauthorized
+	if !c.maywrite {
+		if _, ok := rpcReadOnly[request.Method]; !ok {
+			rpcsLog.Warnf("Read-only websocket client %s tried to access "+
+				"method that changes server's state", c.addr)
+			c.Disconnect()
```

Comment:
This handling is not consistent with the HTTP path which just returns an error.  I think both paths should be consistent, and I'd prefer an error be returned like it is in the HTTP path over disconnected them with no further information.

The reason disconnects are done above is because that path only happens for unauthenticated users who must send the first message as authenticate.